spam/01_spam_tutorial.py

```diff
@@ -192,16 +221,24 @@
 # ### b) Writing an LF to identify spammy comments that use the phrase "check out"
 
 # %% [markdown]
-# Labeling functions in Snorkel are created with the `@labeling_function()` decorator, which wraps a function for evaluating on a single data point (in this case, a row of the DataFrame).
+# Labeling functions in Snorkel are created with the
+# [`@labeling_function()` decorator](https://snorkel.readthedocs.io/en/redux/source/snorkel.labeling.lf.html#snorkel.labeling.lf.core.labeling_function),
+# which wraps a function for evaluating on a single data point.
+# Decorators are a convenient Python mechanism for automatically extending the behavior of the decorated function,
+# and you can learn more [here](https://realpython.com/primer-on-python-decorators/).
 #
 # Let's start developing an LF to catch instances of commenters trying to get people to "check out" their channel, video, or website.
 # We'll start by just looking for the exact string `"check out"` in the text, and compare that to looking for just `"check"` in the text.
-# We'll write a Python function over a single data point (here, a [Pandas `Series` object](https://pandas.pydata.org/pandas-docs/stable/reference/series.html)) to express the two versions of our rule, then add the decorator.
+# We'll write a Python function over a single data point to express the two versions of our rule, then add the decorator.
+# Because our data here is stored as a Pandas DataFrame, a single data point `x` will be a [Pandas `Series` object](https://pandas.pydata.org/pandas-docs/stable/reference/series.html).
+# However, these LFs will work for any object with an attribute named `text`.
```

Comment:
This is a bit confusing. I think saying something like "because of the decorators..." or "The Snorkel API is set up in a way that these LFs can be applied to any object with a text field, such as XYZ". Something to motivate why this is important and that its a part of Snorkel API, not the Pandas API for example